projects/ad9361/src/main.c

```diff
@@ -698,38 +712,225 @@ int main(void)
 	status = no_os_irq_enable(irq_desc, XPAR_FABRIC_AXI_AD9361_DAC_DMA_IRQ_INTR);
 	if(status < 0)
 		return status;
+
+	/* Set callback for MEM2MEM DMAC interrupt. */
+	struct callback_desc mem_dmac_callback = {
+		.ctx = mem_dmac,
+		.callback = axi_dmac_mem_to_mem_isr,
+		.config = NULL
+	};
+
+	status = irq_register_callback(irq_desc,
+				       XPAR_FABRIC_MEM2MEM_DMA_IRQ_INTR, &mem_dmac_callback);
+	if(status < 0)
+		return status;
+
+	status = irq_enable(irq_desc, XPAR_FABRIC_MEM2MEM_DMA_IRQ_INTR);
+	if(status < 0)
+		return status;
 #endif
 
 #ifdef FMCOMMS5
 	axi_dmac_transfer_nonblocking(tx_dmac, DAC_DDR_BASEADDR,
 				      samples * AD9361_ADC_DAC_BYTES_PER_SAMPLE *
 				      (ad9361_phy_b->tx_dac->num_channels + ad9361_phy->tx_dac->num_channels));
 #else
-	axi_dmac_transfer_nonblocking(tx_dmac, (uintptr_t)dac_buffer,
-				      sizeof(sine_lut_iq));
-	no_os_mdelay(1000);
+	/*------------------------------------------------------------------------*/
+	/* Values that can be modified for testing large transfers */
+	//tx_dmac->transfer_max_size = 0x7fff;//must be a multiple of bus width - 1
+	//rx_dmac->transfer_max_size = 0x3fff;
+
+	/* Memory address to write constant data to be transmitted */
+	uint32_t address_mem = 0x4000000;
```

Comment:
We should use some kind of allocated memory.

Reply:
Same as previous comment.